lib/plugins/aws/deploy/lib/extendedValidate.js

```diff
@@ -57,20 +57,29 @@ module.exports = {
       });
     }
 
-    if (
-      !_.isEmpty(this.serverless.service.functions) &&
-      this.serverless.service.package.individually
-    ) {
-      // artifact file validation (multiple function artifacts)
+    if (!_.isEmpty(this.serverless.service.functions)) {
       this.serverless.service.getAllFunctions().forEach(functionName => {
-        let artifactFileName = this.provider.naming.getFunctionArtifactName(functionName);
+        // By default assume service-level package
+        let artifactFileName = this.provider.naming.getServiceArtifactName();
         let artifactFilePath = path.join(this.packagePath, artifactFileName);
 
-        // check if an artifact is used in function package level
         const functionObject = this.serverless.service.getFunction(functionName);
+        const individually =
+          (_.has(functionObject, ['package', 'individually']) &&
+            functionObject.package.individually) ||
+          this.serverless.service.package.individually;
+
         if (_.has(functionObject, ['package', 'artifact'])) {
+          // Use function-level artifact
           artifactFilePath = functionObject.package.artifact;
           artifactFileName = path.basename(artifactFilePath);
+        } else if (this.serverless.service.package.artifact) {
+          // Use service-level artifact
+          artifactFileName = artifactFilePath = this.serverless.service.package.artifact;
+        } else if (individually) {
```

Comment:
If I read correctly, it'll change behavior for configurations in which both `package.individually` and `package.artifact` is set.

Such configuration is obviously contradictory, still currently`package.individually` takes precedence over `package.artifact`, and effect of this PR would be that this precedence will be reverted.

I think it'll be better if we maintain precedence of `package.individually`

Reply:
Good catch! Let me fix it.

Reply:
Good catch. Let me fix it!